kubectl-pg/cmd/scale.go

```diff
@@ -0,0 +1,144 @@
+// Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	PostgresqlLister "github.com/zalando/postgres-operator/pkg/generated/clientset/versioned/typed/acid.zalan.do/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"log"
+	"strconv"
+)
+
+// scaleCmd represents the scale command
+var scaleCmd = &cobra.Command{
+	Use:   "scale",
+	Short: "Add/remove pods to a Postgres cluster",
+	Long: `Scales the postgres objects using cluster-name.
+Scaling to 0 leads to down time.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		clusterName, err := cmd.Flags().GetString("clusterName")
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			log.Fatal(err)
+		}
+		if len(args) > 0 {
+			numberofInstances, err := strconv.Atoi(args[0])
+			if err != nil {
+				log.Fatal(err)
+			}
+			scale(int32(numberofInstances), clusterName, namespace)
+		} else {
+			fmt.Println("Please enter number of instances to scale.")
+		}
+
+	},
+	Example: "kubectl pg scale [NUMBER-OF-INSTANCES] -c [CLUSTER-NAME] -n [NAMESPACE]",
+}
+
+func scale(numberOfInstances int32, clusterName string, namespace string) {
+	config := getConfig()
+	postgresConfig, err := PostgresqlLister.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	postgresql, err := postgresConfig.Postgresqls(namespace).Get(clusterName, metav1.GetOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if numberOfInstances == 0 {
```

Comment:
It makes more sense to put this check after min/max instances check. It may be the case that an operator configuration disallows scaling down to zero

Reply:
Got your point it's more like taking the confirmation and stopping the action after operation validation if scaling to 0 is not possible.